Wikipedia/Code/WMFAuthenticationManager.swift

```diff
@@ -0,0 +1,127 @@
+
+/**
+ *  This class provides a simple interface for performing authentication tasks.
+ */
+class WMFAuthenticationManager: NSObject {    
+    private let keychainCredentials:KeychainCredentials
+    
+    /**
+     *  The current logged in user. If nil, no user is logged in
+     */
+    dynamic private(set) var loggedInUsername: String? = nil
+    
+    /**
+     *  Returns YES if a user is logged in, NO otherwise
+     */
+    public var isLoggedIn: Bool {
+        return (loggedInUsername != nil)
+    }
+
+    private let loginInfoFetcher = WMFAuthLoginInfoFetcher()
+    private let tokenFetcher = WMFAuthTokenFetcher()
+    private let accountLogin = WMFAccountLogin()
+    private let currentlyLoggedInUserFetcher = WMFCurrentlyLoggedInUserFetcher()
+    
+    /**
+     *  Get the shared instance of this class
+     *
+     *  @return The shared Authentication Manager
+     */
+    fileprivate static let _sharedInstance = WMFAuthenticationManager()
```

Comment:
Should I just change this to... 
`public static let sharedInstance = WMFAuthenticationManager()`
... and get rid of the sharedInstance() func? That is, any opposition to or reason not to have the sharedInstance be a property instead of func?